docs/joint_models_estimating_parameters.rst

```diff
@@ -10,7 +10,224 @@ Creating a global hierarchical model
+
+To implement the generalized gamma distribution, we make use of the functionality of scipy’s implementation of the
+distribution. Scipy’s implementation is based on the scientific paper of Stacy (1962) [2]_. An overview of the
+generalized gamma distribution is given in [3]_.
+
+:math:`f(x)=  \frac{k(x-a)^{kc-1}}{b^{kc}\Gamma(c)} \exp \bigg[- \bigg(\frac{x-a}{b}\bigg)^{k}\bigg]`
+
+The Stacy’s generalized gamma distribution involves 4 parameters. Two shape parameters (c, m) , one scale (a) and one
+location parameter (λ). Note, that the above presentedOchi and Stacy’s formulas for the generalized gamma distribution
+differ. Hence, to use scipy’s functionality, we must convert the individual parameters. Comparing the parameters,
+it is seen, that:
+
+1. c=k ,
+2. m=c ,
+3. a=0 , and
+4. λ=1/b
+
+Here, the two shape parameters can be implemented using scipy’s shape parameters. The location parameter will be set to
+the fixed value zero and the scale parameter needs to be converted.
+
+
+**2. Define the characteristics of the distribution and implement it step by step:**
+
+This init method is called when an object of the class GammaDistribution is created. When a new gamma distribution
+object is created, we want to make sure that the attributes of the distribution are passed. Therefore, we build a custom
+initial state, where all parameters of the distribution are initialized with a default value.
+
+.. code-block:: python
+
+       def __init__(self, m=1, c=1, lambda_=1, f_m=None, f_c=None, f_lambda_=None):
+        self.m = m  # shape
+        self.c = c  # shape
+        self.lambda_ = lambda_  # reciprocal scale
+        self.f_m = f_m
+        self.f_c = f_c
+        self.f_lambda_ = f_lambda_
+
+When fitting a distribution function, we want to be able to call the parameters. In virocon, this is ensured by
+“property objects”. A property of an object in Python is a method that seems like a regular attribute to the user
+(i.e. we can use obj.property instead of obj.property() ).
+
+Since scipy uses a slightly different parametrization, than we do here, we need to convert the scale parameter between
+these two parametrizations. For this purpose, we define scale as a property. This allows to calculate the scipy _scale
+on the fly, using our value of _lambda. With the first method we define scale as a property, that we can call as
+x = obj.scale . With the second method, we allow to set the "value" of _scale. (obj._scale = x ) , though again we do
+not store the value directly but instead modify "lambda _" accordingly.
+
+.. code-block:: python
+
+    @property
+    def parameters(self):
+        return {"m": self.m, "c": self.c, "lambda_": self.lambda_}
+
+    @property
+    def _scale(self):
+        return 1 / (self.lambda_)
+
+    @_scale.setter
+    def _scale(self, val):
+        self.lambda_ = 1 / val
+
+Given the case, that we do not know the values of the distribution's parameters, we want to initialize the parameters
```

Comment:
The intent of this function is not initialize parameters not provided. It`s purpose is to convert our parameters to the format scipy understands. It allows to pass values to convert, but if they are None, it uses the distribution instance's current values instead.